base_image_publisher.py

```diff
@@ -0,0 +1,46 @@
+import rospy
+from sensor_msgs.msg import Image
+from cv_bridge import CvBridge, CvBridgeError
+import cv2
+import numpy as np
+
+class Publisher:
+    def __init__(self, cameraport, topic='image_topic', publishrate=0):
+        self.publishrate = publishrate
+        rospy.init_node('image_publisher_node', anonymous=True) # init rosnode
+        self.publisher = rospy.Publisher(topic, Image, queue_size=10) # init topic to publish to
+        self.bridge = CvBridge() # convert opencv image to ros image msg
+        self.cam_feed = cv2.VideoCapture(cameraport) # init device we capture from
+
+    def publish_image(self):
```

Comment:
To support Abdoulaye, we can probably split the publishing process into two function:
1) A function that reads an image from the camera and returns that image as an array and the ret value (maybe def read(self))
2) A function that takes in an image array, converts it to a ROS Image, and publishes it (maybe def publish(self, data).  Abdoulaye can override this method to do his image processing